tests/test_uncheatable_eval_download.py

```diff
@@ -0,0 +1,159 @@
+import importlib.machinery
+import importlib.util
+import sys
+import types
+from pathlib import Path
+
+MODULE_PATH = Path(__file__).resolve().parents[1] / "src" / "marin" / "download" / "uncheatable_eval" / "download.py"
+SPEC = importlib.util.spec_from_file_location("marin.download.uncheatable_eval.download", MODULE_PATH)
+if SPEC is None or SPEC.loader is None:
+    raise RuntimeError("Failed to load uncheatable eval download module for testing")
+
+# Stub heavy optional dependencies before loading the module under test.
+torch_stub = types.ModuleType("torch")
+torch_stub.__dict__.update(
+    {
+        "cuda": types.SimpleNamespace(is_available=lambda: False),
+        "device": lambda *_, **__: None,
+        "__version__": "0.0.0",
+        "Tensor": type("_TensorStub", (), {}),
+        "nn": types.SimpleNamespace(),
+    }
+)
+torch_stub.__spec__ = importlib.machinery.ModuleSpec("torch", loader=None)
+sys.modules.setdefault("torch", torch_stub)
+
+execution_stub = types.ModuleType("marin.execution")
+execution_stub.THIS_OUTPUT_PATH = object()
+execution_stub.ExecutorStep = type("ExecutorStep", (), {})
+execution_stub.VersionedValue = type("VersionedValue", (), {})
+execution_stub.ensure_versioned = staticmethod(lambda value: value)
+execution_stub.this_output_path = staticmethod(lambda: "output")
+sys.modules.setdefault("marin.execution", execution_stub)
```

Comment:
Why do we need to stub everything out here? Can we alternatively structure the code so you don't need to pull these in unless required?